src/app/onion/learning-object-builder/components/content-upload/app/services/file-management.service.ts

```diff
@@ -0,0 +1,234 @@
+      this.S3.upload(
+        params,
+        (err: Error, res: AWS.S3.ManagedUpload.SendData) => {
+          if (err) {
+            this.handleUploadError(err, fileMeta, uploadUpdate$);
+            failed++;
+          } else {
+            const complete: UploadCompleteUpdate = {
+              type: 'complete',
+              data: { ...fileMeta, url: res.Location, eTag: res.ETag }
+            };
+            uploadUpdate$.next(complete);
+            successful++;
+          }
+          remainingUploads--;
+          if (remainingUploads === 0) {
+            uploadUpdate$.next({
+              type: 'queueComplete',
+              data: { successful, failed }
+            });
+          }
+        }
+      ).on('httpUploadProgress', evt => {
+        const progress: UploadProgressUpdate = {
+          type: 'progress',
+          data: {
+            ...fileMeta,
+            progress: (evt.loaded / evt.total) * 100,
+            totalUploaded: evt.loaded
+          }
+        };
+        uploadUpdate$.next(progress);
+      });
+    });
+  }
+
+  /**
+   * Handles errors that occur when uploading file
+   * If error is Credential related; Error is thrown to prevent upload attempts from continuing
+   * All errors are emitted through uploadUpdate
+   *
+   * @private
+   * @param {Error} err [The Error object]
+   * @param {FileUploadMeta} fileMeta [Metadata about the file being uploaded during error]
+   * @param {Subject<UploadUpdate>} uploadUpdate$ [Subject used to emit Upload Error Update]
+   * @memberof FileManagementService
+   */
+  private handleUploadError(
+    err: Error,
+    fileMeta: FileUploadMeta,
+    uploadUpdate$: Subject<UploadUpdate>
+  ) {
+    if (err.name === 'CredentialsError') {
+      const credentialsErrorUpdate: UploadErrorUpdate = {
+        type: 'error',
+        error: err,
+        data: fileMeta
+      };
+      uploadUpdate$.next(credentialsErrorUpdate);
+      throw err;
+    }
+    const errorUpdate: UploadErrorUpdate = {
+      type: 'error',
+      error: err,
+      data: fileMeta
+    };
+    uploadUpdate$.next(errorUpdate);
+  }
+
+  /**
+   * Configures S3 Sdk for upload using user's Cognito credentials
+   *
+   * @private
+   * @memberof FileManagementService
+   */
+  private buildS3Client() {
+    const openIdToken: OpenIdToken = this.auth.getOpenIdToken();
+    const Logins = {};
+    Logins['cognito-identity.amazonaws.com'] = openIdToken.Token;
+    const IdentityPoolId = this.auth.isAdminOrEditor()
+      ? environment.cognitoAdminIdentityPoolId
+      : environment.cognitoIdentityPoolId;
+    const IdentityId = openIdToken.IdentityId;
+    AWS.config.credentials = new AWS.CognitoIdentityCredentials(
+      {
+        IdentityPoolId,
+        IdentityId,
+        Logins
+      },
+      { region: environment.cognitoRegion }
+    );
+    this.S3 = new AWS.S3({ region: environment.s3BucketRegion });
+  }
+
+  /**
+   * Sends learning object ID and file name to API for deletion.
```

Comment:
I'm assuming that the service handles the file deletion still. Maybe now that we have a mix of client and server accessing S3, the JSDoc could specify that this is what happens, and so we don't need to manually connect to S3 for this.